src/Controller/ShopwareFilesController.php

```diff
@@ -0,0 +1,188 @@
+<?php declare(strict_types=1);
+
+namespace Frosh\Tools\Controller;
+
+use Shopware\Core\Kernel;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+#[Route(path: '/api/_action/frosh-tools', defaults: ['_routeScope' => ['api'], '_acl' => ['frosh_tools:read']])]
+class ShopwareFilesController
+{
+    private const STATUS_OK = 0;
+    private const STATUS_IGNORED_ALL = 1;
+    private const STATUS_IGNORED_IN_PROJECT = 2;
+    private bool $isPlatform;
+
+    public function __construct(
+        private readonly string $shopwareVersion,
+        private readonly string $projectDir,
+        private readonly array $projectIgnoredFiles
+    )
+    {
+        $this->isPlatform = !is_dir($this->projectDir . '/vendor/shopware/core') && is_dir($this->projectDir . '/src/Core');
+    }
+
+    #[Route(path: '/shopware-files', name: 'api.frosh.tools.shopware-files', methods: ['GET'])]
+    public function listShopwareFiles(): JsonResponse
+    {
+        if ($this->shopwareVersion === Kernel::SHOPWARE_FALLBACK_VERSION) {
+            return new JsonResponse(['error' => 'Git version is not supported']);
+        }
```

Comment:
we should move that into an private method as it's called here

Reply:
as discussed: these are reponses. this would not make any better.